File: server/Models/Post.cs
using System;
using System.Text.Json.Serialization;

namespace server.Models
{

    public class Post{
        public string Id {get; set;} = Guid.NewGuid().ToString();
        public required string AuthorId {get; set;} = string.Empty;
        public required string Content {get; set;} = string.Empty;
        public DateTime CreatedAt {get; set;} = DateTime.Now;
        public string Username {get; set;} = string.Empty;
    }

}

File: client/src/app/Models/Post.ts
import { Like } from "./like";
import { Comment } from "./comment";
export interface Post {
    id: string;
    authorId: string;
    content: string;
    imageUrl?: string | null;
    imageKey?: string | null;
    createdAt: Date;
    username?: string;
    likes?: Like[];
    likedByCurrentUser?: boolean;
}
